MapboxCoreNavigation/FreeDriveLocationManager.swift

```diff
@@ -0,0 +1,134 @@
+import MapboxNavigationNative
+import CoreLocation
+import MapboxDirections
+import MapboxAccounts
+
+public protocol FreeDriveDebugInfoListener: AnyObject {
+    func didGet(location: CLLocation, with matches: [MapMatch], for rawLocation: CLLocation)
+}
+
+open class FreeDriveLocationManager: NavigationLocationManager, CLLocationManagerDelegate {
+    public weak var debugInfoListener: FreeDriveDebugInfoListener? {
+        get {
+            proxyDelegate?.debugDelegate
+        }
+        set {
+            proxyDelegate?.debugDelegate = newValue
+        }
+    }
+
+    private var proxyDelegate: ProxyDelegate?
+
+    public required init(navigator: Navigator? = nil) {
```

Comment:
MapboxNavigationNative types should not be part of the public API. The navigator is an implementation detail, and its interface is written in Objective-C.